hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -205,11 +205,17 @@ protected void onRemoveError(K key, R record, boolean removed, Throwable error)
 
     @SuppressWarnings("unused")
     protected void onExpire(K key, R record) {
+        if (!canUpdateStats(record)) {
```

Comment:
Right now, we have the same check here and in `HeapNearCacheRecordMap`. Do we need to have both of them?